### PR TITLE
Arm64/Sve: Implement Compute*BitAddresses APIs

### PR DESCRIPTION
**Pull request by kunalspathak**
Implements following APIs:

- Compute16BitAddresses
- Compute32BitAddresses
- Compute64BitAddresses
- Compute8BitAddresses

Contributes towards https://github.com/dotnet/runtime/issues/99957

### CODE REVIEW

**Review by TIHan — APPROVED** [review]
LGTM pending test results and the formatting fixes

File: src/coreclr/jit/hwintrinsiccodegenarm64.cpp [hunk 1]
```diff
@@ -1937,7 +1937,26 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 }
                 break;
             }
-
+            case NI_Sve_Compute16BitAddresses:
```
Comment (a74nh) [inline src/coreclr/jit/hwintrinsiccodegenarm64.cpp:1940]: I'm tempted to suggested all these cases should be combined and the constant should be: `( id -NI_Sve_Compute8BitAddresses )`. With an assert to confirm `NI_Sve_Compute8BitAddresses` to `NI_Sve_Compute64BitAddresses` is sequential.

But also happy as it is

Comment (kunalspathak) [reply src/coreclr/jit/hwintrinsiccodegenarm64.cpp:1940]: yes, was not happy with 4 different case statements. Thanks for the suggestion.

**Comment by kunalspathak** [comment]
Stress test pass except the failure in JitStress because of known preodicate registers. https://gist.github.com/kunalspathak/19eb5656261d3978843927b746788da6

**Comment by kunalspathak** [comment]
/ba-g superpmi replay failures https://github.com/dotnet/runtime/issues/102773

**Pull request merged**